### PR TITLE
Chat remodel

### PR DESCRIPTION
**Pull request by nyashafrank**
Changed the way chat calls are structured. Made use of the action creators, but subscribeToRoom() is still within the direct_messages.js component because it deals with hooks and I'm not entirely sure how to transform that into an action.

There was a problem where chatkit tried connecting to not logged in users. Also, a problem when a user (not logged in) views a service, then logs in, tries to message the seller, but their chat connection was not updated. But, thanks to ComponentDidUpdate(), we good. I'm so sorry this took foreverrr to fix, I'm literally an idiot and kept getting in my own way.  also, sorry for the mumbo jumbo pieces of code from before.

Fixing the merge conflicts now. New year new me

### CODE REVIEW

**Comment by nyashafrank** [comment]
oh I also want to toggle the login modal when a user not logged in clicks message seller

**Comment by nyashafrank** [comment]
how we do that lol

**Comment by abeSanchez** [comment]
lookin thru

**Comment by abeSanchez** [comment]
but yeah i wanted to do that too but i left the button disabled for now if the user is not logged in or if the user thats logged in is the owner of that service

File: src/actions/index.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import axios from 'axios';
+import { instanceLocator, tokenUrl } from '../components/direct_message_config';
```
Comment (abeSanchez) [inline src/actions/index.js]: are you using these in here? also could you take this out of the components and put it in the constants file pls? thanks

File: src/actions/index.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
...
+import { ChatManager, TokenProvider } from '@pusher/chatkit-client';
```
Comment (abeSanchez) [inline src/actions/index.js]: Do you need these here?

File: src/components/service_view.js [hunk 3]
```diff
@@ -52,21 +45,13 @@ class ServiceView extends Component {
 
         this.props.fetchServiceAndOwner(id);
 
-        if (!this.props.loggedIn) return;
```
Comment (abeSanchez) [inline src/components/service_view.js]: rude

**Comment by abeSanchez** [comment]
what the problem is tho

**Comment by abeSanchez** [comment]
lol my bad i cant read. alright cool

File: src/actions/index.js (hunk 2 above)
Comment (nyashafrank) [reply src/actions/index.js]: Yes, not sure where else they would go

File: src/components/service_view.js (hunk 3 above)
Comment (nyashafrank) [reply src/components/service_view.js]: 👋

File: src/actions/index.js (hunk 1 above)
Comment (nyashafrank) [reply src/actions/index.js]: Moved :)

File: src/actions/index.js (hunk 2 above)
Comment (abeSanchez) [reply src/actions/index.js]: i only ask cuz theyre not being used

**Comment by nyashafrank** [comment]
@steevejoseph is it cool if I delete the chat branch, but keep chat-remodel up

File: src/actions/index.js (hunk 2 above)
Comment (nyashafrank) [reply src/actions/index.js]: lines 223-227 tho

**Pull request merged**